src/main/java/com/leverx/employeestat/rest/controller/EmployeeController.java

```diff
@@ -17,46 +22,59 @@
 public class EmployeeController {
 
     private final EmployeeService employeeService;
-    private final EmployeeConverter employeeConverter;
+    private final BindingResultParser bindingResultParser;
 
     @Autowired
-    public EmployeeController(EmployeeService employeeService, EmployeeConverter employeeConverter) {
+    public EmployeeController(EmployeeService employeeService, BindingResultParser bindingResultParser) {
         this.employeeService = employeeService;
-        this.employeeConverter = employeeConverter;
+        this.bindingResultParser = bindingResultParser;
     }
 
     @GetMapping
     @ResponseStatus(HttpStatus.OK)
     public List<EmployeeDTO> getAllEmployees() {
-        return employeeService.getAll()
-                .stream()
-                .map(employeeConverter::toDTO)
-                .collect(Collectors.toList());
+        return employeeService.getAll();
     }
 
     @GetMapping("/{id}")
     @ResponseStatus(HttpStatus.OK)
-    public EmployeeDTO getEmployee(@PathVariable UUID id) {
-        return employeeConverter.toDTO(employeeService.getById(id));
+    public EmployeeDTO getEmployee(@PathVariable String id) {
+        return employeeService.getById(getUUIDFromString(id));
     }
 
     @PostMapping
     @ResponseStatus(HttpStatus.OK)
-    public EmployeeDTO postEmployee(@RequestBody EmployeeDTO employeeDTO) {
-        Employee employee = employeeConverter.toEntity(employeeDTO);
-        return employeeConverter.toDTO(employeeService.save(employee));
+    public EmployeeDTO postEmployee(@RequestBody @Valid EmployeeDTO employeeDTO, BindingResult result) {
+        if (result.hasErrors()) {
+            throw new NotValidRecordException("Fields of Employee have errors: " +
+                    bindingResultParser.getFieldErrMismatches(result));
+        }
+        return employeeService.save(employeeDTO);
     }
 
     @PutMapping
     @ResponseStatus(HttpStatus.OK)
-    public EmployeeDTO putEmployee(@RequestBody EmployeeDTO employeeDTO) {
-        Employee employee = employeeConverter.toEntity(employeeDTO);
-        return employeeConverter.toDTO(employeeService.update(employee));
+    public EmployeeDTO putEmployee(@RequestBody @Valid EmployeeDTO employeeDTO, BindingResult result) {
+        if (result.hasErrors()) {
+            throw new NotValidRecordException("Fields of Employee have errors: " +
+                    bindingResultParser.getFieldErrMismatches(result));
+        }
+        return employeeService.update(employeeDTO);
     }
 
     @DeleteMapping("/{id}")
     @ResponseStatus(HttpStatus.OK)
-    public void deleteEmployee(@PathVariable UUID id) {
-        employeeService.deleteById(id);
+    public void deleteEmployee(@PathVariable String id) {
+        employeeService.deleteById(getUUIDFromString(id));
+    }
+
+    private UUID getUUIDFromString(String id) {
```

Comment:
same here, duplicated method from other controller, so it has sense to extract it